frege/compiler/common/Mangle.fr

```diff
@@ -3,21 +3,24 @@ module frege.compiler.common.Mangle where
 
 import Data.TreeMap as TM
 
+-- Generates an ident which is guaranteed not to clash with any user defined ident.
+noClashIdent :: String -> String
+noClashIdent i = "$" ++ i
 
 {--
- * Java keywords (along with substitutions)
- * as specified in the Java Language Specification, Version 3.0, page 21
+ * Java keywords (along with substitutions) as specified in the
+ * [Java Language Specification, Version 8.0](https://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html#jls-3.9)
  * and, in addition, "true", "false", and "null"
  *
  * It is, for instance, possible to name a frege item _int_, though
- * in the generated java code, this will appear as \__int_.
+ * in the generated java code, this will appear as \$_int_.
  *
  -}
-javakeywords = TM.fromList  [(kw, "_"++kw) | kw <- [
+javakeywords = TM.fromList  [(kw, noClashIdent kw) | kw <- [
     "abstract",     "continue",     "for",          "new",          "switch",
     "assert",       "default",      "if",           "package",      "synchronized",
     "boolean",      "do",           "goto",         "private",      "this",
-    "break",        "double",       "implements",   "protected",    "then",
```

Comment:
I'm not sure what I was thinking *then* (pun intented), but a frege identifier can hardly be `then` anyway ...